### ISSUE TITLE
tilelive-vector not passing due to failed s3 auth

### ISSUE THREAD

**Issue by mapsam** [mapbox/mapnik-swoop#9]
A recent [commit to `tilelive-vector`](https://github.com/mapbox/tilelive-vector/commit/384231fe72084b7e20b19f6f1634cc74a1253065) shows some required credentials that we don't have when running local or travis tests, which means we are failing on the swoop end of things. Maybe @willwhite can point us in the proper direction to get things working?

Here is the [swoop error](https://travis-ci.org/mapbox/mapnik-swoop/jobs/100902476):

```
tilelive-vector --> 
waiting...
waiting...
waiting...
waiting...
waiting...
waiting...
waiting...
waiting...
waiting...
waiting...
waiting...
waiting...
waiting...
not ok 1214 CredentialsError: Missing credentials in config
  ---
    operator: error
    expected:
      undefined
    actual:
      { code: 'CredentialsError', message: 'Missing credentials in config', originalError: { code: 'CredentialsError', message: 'Could not load credentials from any providers', originalError: { message: 'Unexpected token <' }, time: Wed Jan 20 2016 19:48:10 GMT+0000 (UTC) }, time: Wed Jan 20 2016 19:48:10 GMT+0000 (UTC) }
    at: PassThrough.error (/home/travis/build/mapbox/mapnik-swoop/node_modules/tilelive-vector/index.js:440:46)
    stack:
      CredentialsError: Missing credentials in config
        at Object.parse (native)
        at /home/travis/build/mapbox/mapnik-swoop/node_modules/tilelive-vector/node_modules/aws-sdk/lib/metadata_service.js:115:38
        at IncomingMessage.<anonymous> (/home/travis/build/mapbox/mapnik-swoop/node_modules/tilelive-vector/node_modules/aws-sdk/lib/metadata_service.js:74:45)
        at IncomingMessage.emit (events.js:117:20)
        at _stream_readable.js:944:16
        at process._tickDomainCallback (node.js:492:13)
  ...
# errors out on tm2z file on s3 where we do not have access
not ok 1215 should be equal
  ---
    operator: equal
    expected: 'AccessDenied'
    actual:   'CredentialsError'
    at: PassThrough.error (/home/travis/build/mapbox/mapnik-swoop/node_modules/tilelive-vector/index.js:440:46)
  ...
# invalid
waiting...
np
```

**Comment by willwhite** [comment]
Hm interesting problem. Should we make it possible for this project to skip running those tilelive-vector tests that require AWS credentials? They are specific to tilelive-vector intergration with S3 and related to mapnik. I'm not sure if this project should aim to clone all of the test AWS privileges that dependent modules have.

**Comment by springmeyer** [comment]
@willwhite skipping tests that don't invoke mapnik would be just fine. Maybe we could accomplish that with an something like `TILELIVE_VECTOR_NO_TEST_AUTH=1 npm test` and when that is set the `tilelive-vector` tests would skip things intelligently?

**Comment by mapsam** [comment]
I agree, adding the s3 creds seems a little too specific for `mapnik-swoop`, especially because they could change at some point. 

@springmeyer would that require us to update in `tilelive-vector` or within swoop itself?

**Comment by springmeyer** [comment]
@mapsam - would require an update in `tilelive-vector` mainly - I can't think of a way to avoid that.

**Comment by mapsam** [comment]
Sounds good. I'm trying to think of ways to make the `tilelive-vector` tests skip without being specific to `mapnik-swoop`. Passing `TILELIVE_VECTOR_NO_TEST_AUTH` seems like it should work! Also the tests in `tilelive-vector` could be written to ONLY test the s3 call IF credentials exist, otherwise move one. Maybe that defeats the point of the tests though.

I'm going to hold on this now, in order to not fall too deep down a rabbit hole.

**Comment by mapsam** [comment]
First pass at a workaround here: https://github.com/mapbox/tilelive-vector/pull/116

**Comment by springmeyer** [comment]
Sweet, merged, and fixed up in mapnik-swoop in 1b02055
